superset-frontend/src/explore/components/DisplayQueryButton/DisplayQueryButton.test.tsx

```diff
@@ -0,0 +1,230 @@
+    datasource: 'FCC 2018 Survey',
+    description: null,
+    description_markeddown: '',
+    edit_url: '/chart/edit/318',
+    form_data: {
+      adhoc_filters: [],
+      all_columns_x: ['age'],
+      color_scheme: 'supersetColors',
+      datasource: '49__table',
+      granularity_sqla: 'time_start',
+      groupby: null,
+      label_colors: {},
+      link_length: '25',
+      queryFields: { groupby: 'groupby' },
+      row_limit: 10000,
+      slice_id: 318,
+      time_range: 'No filter',
+      time_range_endpoints: ['inclusive', 'exclusive'],
+      url_params: {},
+      viz_type: 'histogram',
+      x_axis_label: 'age',
+      y_axis_label: 'count',
+    },
+    modified: '<span class="no-wrap">3 days ago</span>',
+    owners: [],
+    slice_id: 318,
+    slice_name: 'Age distribution of respondents',
+    slice_url: '/superset/explore/?form_data=%7B%22slice_id%22%3A%20318%7D',
+  },
+  chartStatus: 'rendered',
+  onOpenPropertiesModal: jest.fn(),
+  onOpenInEditor: jest.fn(),
+};
+
+beforeEach(() => {
+  props = {
+    latestQueryFormData: {
+      viz_type: 'histogram',
+      datasource: '49__table',
+      slice_id: 318,
+      url_params: {},
+      time_range_endpoints: ['inclusive', 'exclusive'],
+      granularity_sqla: 'time_start',
+      time_range: 'No filter',
+      all_columns_x: ['age'],
+      adhoc_filters: [],
+      row_limit: 10000,
+      groupby: null,
+      color_scheme: 'supersetColors',
+      label_colors: {},
+      link_length: '25',
+      x_axis_label: 'age',
+      y_axis_label: 'count',
+    },
+    slice: {
+      cache_timeout: null,
+      changed_on: '2021-03-19T16:30:56.750230',
+      changed_on_humanized: '3 days ago',
+      datasource: 'FCC 2018 Survey',
+      description: null,
+      description_markeddown: '',
+      edit_url: '/chart/edit/318',
+      form_data: {
+        adhoc_filters: [],
+        all_columns_x: ['age'],
+        color_scheme: 'supersetColors',
+        datasource: '49__table',
+        granularity_sqla: 'time_start',
+        groupby: null,
+        label_colors: {},
+        link_length: '25',
+        queryFields: { groupby: 'groupby' },
+        row_limit: 10000,
+        slice_id: 318,
+        time_range: 'No filter',
+        time_range_endpoints: ['inclusive', 'exclusive'],
+        url_params: {},
+        viz_type: 'histogram',
+        x_axis_label: 'age',
+        y_axis_label: 'count',
+      },
+      modified: '<span class="no-wrap">3 days ago</span>',
+      owners: [],
+      slice_id: 318,
+      slice_name: 'Age distribution of respondents',
+      slice_url: '/superset/explore/?form_data=%7B%22slice_id%22%3A%20318%7D',
+    },
+    chartStatus: 'rendered',
+    onOpenPropertiesModal: jest.fn(),
+    onOpenInEditor: jest.fn(),
+  };
+});
+
+test('Shoud render a button', () => {
+  render(<DisplayQueryButton {...props} />, { useRedux: true });
+  expect(screen.getByRole('button')).toBeInTheDocument();
+});
+
+test('Shoud open a menu', () => {
```

Comment:
Here and in the other tests you can mock the listeners instead of using `props`. 
```
const onOpenInEditor = jest.fn();
const onOpenPropertiesModal = jest.fn();
render(<ConnectedDisplayQueryButton
   {...props}
   onOpenInEditor={onOpenInEditor}
   onOpenPropertiesModal={onOpenPropertiesModal}
 />,
 {
   useRedux: true,
 },
);
expect(onOpenInEditor).toBeCalledTimes(0);
userEvent.click(screen.getByRole('button'));
userEvent.click(screen.getByRole('menuitem', { name: 'Edit properties' }));
expect(onOpenPropertiesModal).toBeCalledTimes(1);
```
This way you can remove all the duplicated code in `beforeEach` and change `props` to a `const`. I tested locally and the props will not be modified. It will greatly reduce the number of lines 😉

Reply:
You can use `fetchMock` to resolve the `Unmatched POST to http://api/v1/chart/data?form_data=%7B%22slice_id%22%3A318%7D` warning